### DISCUSSION TITLE
Defining an ROI to Restrict Sampling

### DISCUSSION THREAD

**Discussion by BraydenSchott** [MIC-DKFZ/nnUNet#577] [Q&A] [answered]
Hi There,

Thank you for sharing "nnUNet" on GitHub. Is there any way we can provide the model with a pre-defined ROI to restrict the data sampling to regions within the ROI? I realize this may not make a ton of sense given the patch size is very large. Perhaps I can simply crop my images to the pre-defined ROI and filter out any out-of-ROI probabilities as a post-processing step.

Also, there seems to be an option in the network configuration to define a mask for normalization as defined by the parameter "use_mask_for_norm" parameter listed in the config summary prior to initiating the training. I don't see this parameter in the setup "dataset.json" example file. Is this parameter settable? 

Thank You,

-Brayden

**Comment by FabianIsensee** [comment, marked as answer]
Hi Brayden,

this mask is not settable. It is intended for images where the object of interest (like the brain) is surrounded by black background pixels. For those images it is common practice to only use the brain region for normalization.

If you want to only train on specific ROIS, the best way to do that would be to crop the images to the desired ROI before you give them to nnU-Net (as you suggested). Mote that nnU-Net will then only be able to make predictions in the parts that are covered by the ROIs, so you will need to also crop all the images you intend to apply the models to.

Best,
Fabian

**Reply by BraydenSchott** [reply]
Thank you, Fabian. Helpful reply and great repo you provide.

**Reply by HuiLin0220** [reply]
> Hi Brayden,
> 
> this mask is not settable. It is intended for images where the object of interest (like the brain) is surrounded by black background pixels. For those images it is common practice to only use the brain region for normalization.
> 
> If you want to only train on specific ROIS, the best way to do that would be to crop the images to the desired ROI before you give them to nnU-Net (as you suggested). Mote that nnU-Net will then only be able to make predictions in the parts that are covered by the ROIs, so you will need to also crop all the images you intend to apply the models to.
> 
> Best, Fabian

Is the input in the training set just the brain regions cropped from the original image? 
'If cropping results in an average size decrease of 25% or more, a mask for central non-zero voxels is created, and normalization is applied within that mask only, ignoring the surrounding zero voxels' in your paper. If the brain region is very large (80%), is normalization applied in the original image?

**Reply by FabianIsensee** [reply]
if we do cropping then this is the procedure:

- crop to nonzero area
- create a mask of the nonzero area
- normalize within the nonzero area only using the mask, leave the mask at 0

This is done during the preprocessing and will also be done to your test cases when running inference
